observation_portal/requestgroups/serializers.py

```diff
@@ -421,6 +450,35 @@ def validate(self, data):
                             ))
                         )
 
+            # Validate the exposure modes
+            if 'exposure' in modes:
+                exposure_mode_validation_helper = ModeValidationHelper('exposure', instrument_type, default_modes,
+                                                                       modes)
+                if exposure_mode_validation_helper.mode_is_not_set(instrument_config['extra_params']):
+                    exposure_mode_to_set = exposure_mode_validation_helper.get_mode_to_set()
+                    if exposure_mode_to_set['error']:
+                        raise serializers.ValidationError(_(rotator_mode_to_set['error']))
+                    if exposure_mode_to_set['mode']:
+                        instrument_config['extra_params']['exposure_mode'] = exposure_mode_to_set['mode']['code']
+
+                exposure_mode_error_msg = exposure_mode_validation_helper.get_mode_error_msg(
+                    instrument_config['extra_params']['exposure_mode'],
+                    instrument_config
+                )
+                if exposure_mode_error_msg:
+                    raise serializers.ValidationError(_(exposure_mode_error_msg))
+
+            # Validate the exposure times for MUSCAT are positive numbers
```

Comment:
This extra stuff is to validate the exposure_time_* are non-negative numbers. We should maybe make this type of validation automatic in the ModeValidationHelper by defined a schema to define type of required fields in configdb, like `{"required_fields": ["exposure_time_g[float]", "exposure_mode[string]"]}` or something like that